pnet_transport/src/lib.rs

```diff
@@ -269,6 +281,19 @@ impl TransportSender {
         set_sockopt_u8(self.socket.clone(), level, name, tos as u8)
     }
 
+    /// Configures path MTU discovery on a socket.
+    ///
+    /// On Linux-based systems, configuring the socket's path MTU discovery setting is
+    /// the way to "(un)set" the "don't fragment" bit in the IP header.
+    #[cfg(target_os = "linux")]
+    pub fn configure_path_mtu_discovery(&mut self, enable_disable: bool) -> io::Result<()> {
+        let (level, name, value) = match self.channel_type {
+            Layer4(Ipv4(_)) | Layer3(_) => (pnet_sys::IPPROTO_IP, pnet_sys::IP_MTU_DISCOVER, if enable_disable { pnet_sys::IP_PMTUDISC_DO } else {pnet_sys::IP_PMTUDISC_DONT  }),
+            Layer4(Ipv6(_)) => (pnet_sys::IPPROTO_IPV6, pnet_sys::IPV6_MTU_DISCOVER, if enable_disable { pnet_sys::IPV6_PMTUDISC_DO } else {pnet_sys::IPV6_PMTUDISC_DONT  }),
+        };
```

Comment:
This match statement is a bit intense :wink: 

Please reformat or refactor this so it's easier to read.

Reply:
Absolutely! I think that I got a little carried away!!

Reply:
I have made a two-dimensional array that holds the different `level`, `name`, `value` combinations and use the `self.channel_type` and `enable_disable` to generate indexes. This seemed to me like a reasonable tradeoff between making that monstrous match (alliteration!) more readable and minimizing code duplication. If you still think that it is too unreadable, please let me know -- I'd be glad to take another swing at it.